packages/fxa-payments-server/src/lib/AppLocalizationProvider.tsx

```diff
@@ -73,47 +82,46 @@ export default class AppLocalizationProvider extends Component<Props, State> {
     baseDir: '/locales',
     userLocales: ['en-US'],
     bundles: ['main'],
-    children: React.createElement('div')
+    children: React.createElement('div'),
+  };
+
+  constructor(props: Props) {
+    super(props);
+    const { baseDir, userLocales, bundles } = props;
+
+    this.state = {
+      baseDir,
+      userLocales,
+      bundles,
+    };
   }
 
-   constructor(props: Props) {
-     super(props);
-     const { baseDir, userLocales, bundles } = props;
-
-     this.state = {
-       baseDir,
-       userLocales,
-       bundles,
-     };
-   }
-
-   async componentDidMount() {
-     const { baseDir, userLocales, bundles } = this.state;
-
-     const availableLocales = await fetchAvailableLocales(baseDir);
-     const currentLocales = negotiateLanguages(
-       userLocales, availableLocales,
-       { defaultLocale: availableLocales[0] }
-     );
-
-     const generateMessages = await createMessagesGenerator(
-       baseDir, currentLocales, bundles
-     );
-     this.setState({ messages: generateMessages() });
-   }
-
-   render() {
-     const { children } = this.props;
-     const { messages } = this.state;
-
-     if (!messages) {
-       return <div/>;
-     }
-
-     return (
-       <LocalizationProvider bundles={messages}>
-         {children}
-       </LocalizationProvider>
-     );
-   }
- }
+  async componentDidMount() {
+    const { baseDir, userLocales, bundles } = this.state;
+
+    const availableLocales = await fetchAvailableLocales(baseDir);
+    const currentLocales = negotiateLanguages(userLocales, availableLocales, {
+      defaultLocale: availableLocales[0],
+    });
+
+    const generateMessages = await createMessagesGenerator(
+      baseDir,
+      currentLocales,
+      bundles
+    );
+    this.setState({ messages: generateMessages() });
```

Comment:
I've had some issues storing the generator in the state like this. In fact, that's the main reason behind the change to `LocalizationProvider`'s API in https://github.com/projectfluent/fluent.js/pull/461. In particular, in one the apps I was testing in, it seemed like React wasn't always memoizing the provider and instead would re-render it using the same bundles generator. And since the generator keeps its own state of iteration, multiple re-renders would deplete it :( Have you seen anything like this?

In any case, I'll be happy to file a PR to upgrade to the new `LocalizationProvider` API once I finish working on it.

Reply:
That sounds great, thank you!